addons/mail/static/tests/legacy/discuss/core/web/chat_window_new_message_tests.js

```diff
@@ -147,3 +147,27 @@ QUnit.test("new message autocomplete should automatically select first result",
     await insertText(".o-discuss-ChannelSelector input", "131");
     await contains(".o-discuss-ChannelSelector-suggestion a.o-mail-NavigableList-active");
 });
+
+QUnit.test('open chat from "new message" chat window should unfold existing window', async () => {
```

Comment:
Note: it's nice to have this test (to cover the JS part).
But the test actually doesn't cover the fix at all: the fix is mostly done in python, and mock server hasn't been adapted accordingly. And it would make no difference to adapt it, because the issue is a very specific race condition anyway.

Reply:
I did not change the implementation but updated the function signature